src/renderTemplate.tsx

```diff
@@ -1,60 +1,37 @@
-import cons from "consolidate"
-import { isArray, isFunction } from "lodash"
+import { isArray } from "lodash"
 import path from "path"
 import { StitchOptions } from "./index"
-import { Block } from "./render"
+import { getCompileFn } from "./utils"
 
 type RenderTemplateOptions = Pick<
   StitchOptions,
   "basePath" | "locals" | "config"
 >
 
 export async function renderTemplate(
-  block: Block,
+  filePaths: string | string[],
   options: RenderTemplateOptions = {}
 ): Promise<string> {
-  const {
-    basePath = process.cwd(),
-    locals = {},
-    config = {
-      engines: {},
-    },
-  } = options
+  const { config, basePath = process.cwd() } = options
 
   try {
-    const rendered = isArray(block)
-      ? await Promise.all(block.map(compile))
-      : await compile(block)
+    const rendered = isArray(filePaths)
+      ? await Promise.all(filePaths.map(compile)).then(res => res.join("\n"))
```

Comment:
@damassi Is this option of passing in an array of templates actually exercised in any of our code or is this a case of YAGNI? Because the `renderSwitch` call-site was expecting this to be single string and `renderTemplate` is typed to return a single string too.

Reply:
The reason this previously passed is that the `cons[ext]` was typed as `any`, thus matching whatever return type you specified.

Reply:
```
cons[ext]
```
can pull from any number of [consolidate.js](https://github.com/tj/consolidate.js/) engines. 

I haven't spent any time making the typings exact and used any when I ran into a blocker. I slipped in this typescript refactor when I had a day but haven't returned to it. We can return to this unless you feel like you can tidy up.

Reply:
Oh, I don’t feel strongly at all about the typings, what I am wondering about is if the arrays version is being used at all [and if I’m not breaking its usage with this join].